### DISCUSSION TITLE
Add support for UUIDv5

### DISCUSSION THREAD

**Discussion by kuzmi4** [directus/directus#13278] [Feature Requests]
### Describe the Bug

If upload file with ID field in multiform data, file will upload fine, but will not be accessible (403 http code)

For generate ID used [UUID lib](https://www.npmjs.com/package/uuid)
I tested with local, and S3 storage driver. Value in "filename_disk" same as filename in dir/bucket. Files readable from fs/s3

![image](https://user-images.githubusercontent.com/6544815/168156779-4b202c20-e4fa-41d1-9d69-7bb7000a156c.png)

![screen 2022-05-12 в 22 50 37](https://user-images.githubusercontent.com/6544815/168157181-c6c92547-12aa-4be9-afd3-d95bc1360ccc.png)

### To Reproduce

Upload file with ID filed in multiform data

### Errors Shown

{"errors":[{"message":"You don't have permission to access this.","extensions":{"code":"FORBIDDEN"}}]}

### What version of Directus are you using?

9.10.0

### What version of Node.js are you using?

16.15.0

### What database are you using?

Postgres 14

### What browser are you using?

Edge

### How are you deploying Directus?

docker

**Comment by br41nslug** [comment]
I am unable to reproduce it locally using docker, postgres 14 on v9.10.0
Are you sure this is not an issue permissions?

I'll close this for now, but we'll reopen when more information becomes available.

**Comment by kuzmi4** [comment]
So, i create demo stand in directus cloud https://5lcx4hlg.directus.app/ and record video

https://user-images.githubusercontent.com/6544815/168296582-812bc785-8e45-4216-9081-e0b48a6d1dee.mov



Use token "demo" for request:

I using next request for problem reproduce:
```
curl --location --request POST 'https://5lcx4hlg.directus.app/files' \
--header 'Authorization: Bearer demo' \
--form 'id="27bde3e0-deed-52a7-bf72-054811872c29"' \
--form 'title="demo123"' \
--form 'description="demo456"' \
--form 'tags="[004]"' \
--form 'file=@"/Users/kuzmi4/Desktop/Screens/screen 2022-05-12 в 22.50.37.png"'
```

**Comment by br41nslug** [comment]
What i think is happening here is that you are using UUIDv5 while Directus uses UUIDv4 which don't seem compatible even tho they have the exact same structure.
Can you make sure you are generating UUIDv4?

**Comment by kuzmi4** [comment]
You are right, with V4 file is accessible.

I use V5 for ID control, and it`s easiest way for me, is it possible expand directus check for all uuid formats?

**Reply by br41nslug** [reply]
I think postgreSQL is currently the blocking factor for this as directus uses the native UUID type which only supports valid UUIDv4 currently

**Reply by kuzmi4** [reply]
I use Postgres database, and generate UUID manually for all items. All work fine.

For be clear, all data IDs in non directus tables - has format  uuid v5.

**Reply by kuzmi4** [reply]
How i see in API used lib: https://www.npmjs.com/package/uuid-validate for UUID validation 
and file assets check only V4 version:
https://github.com/directus/directus/blob/cc076494ff9ddefec97741fda281ef55922451fe/api/src/services/assets.ts#L52

It can be replace by https://www.npmjs.com/package/uuid#uuidvalidatestr for UUID validation

**Reply by br41nslug** [reply]
You're very right! Thanks for the PR will try to review it today :)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
